### PR TITLE
V1.7.16 patchset

### PR DESCRIPTION
**Pull request by rphillips**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

**Review by ericchiang — APPROVED** [review]
Lgtm

File: cluster/images/hyperkube/Makefile [hunk 1]
```diff
@@ -21,7 +21,7 @@ REGISTRY?=gcr.io/google_containers
 ARCH?=amd64
 OUT_DIR?=_output
 TEMP_DIR:=$(shell mktemp -d -t hyperkubeXXXXXX)
-CNI_RELEASE=0799f5732f2a11b329d9e3d51b9c8f2e3759f2ff
+CNI_RELEASE=v0.6.0
```
Comment (colemickens) [inline cluster/images/hyperkube/Makefile:24]: Why this change? CNI and Kubernetes versions have a matrix of compatibility, especially for old versions like 1.7.x.

Or is this a known good revision to use with 1.7.x?

File: cluster/images/hyperkube/Makefile [hunk 2]
```diff
@@ -61,7 +61,7 @@ ifndef VERSION
     $(error VERSION is undefined)
 endif
 	cp -r ./* ${TEMP_DIR}
-	mkdir -p ${TEMP_DIR}/cni-bin
+	mkdir -p ${TEMP_DIR}/cni-bin/bin
```
Comment (colemickens) [inline cluster/images/hyperkube/Makefile:64]: I guess a result of a change that happened with CNI packaging due to the bump?

File: cluster/images/hyperkube/Makefile [hunk 3]
```diff
@@ -84,7 +84,10 @@ else
 	curl -sSL --retry 5 https://github.com/multiarch/qemu-user-static/releases/download/$(QEMUVERSION)/x86_64_qemu-${QEMUARCH}-static.tar.gz | tar -xz -C ${TEMP_DIR}
 endif
 	# Download CNI
-	curl -sSL --retry 5 https://storage.googleapis.com/kubernetes-release/network-plugins/cni-${ARCH}-${CNI_RELEASE}.tar.gz | tar -xz -C ${TEMP_DIR}/cni-bin
+	curl -sSL --retry 5 https://github.com/containernetworking/cni/releases/download/${CNI_RELEASE}/cni-${ARCH}-${CNI_RELEASE}.tgz | tar -xz -C ${TEMP_DIR}/cni-bin/bin
```
Comment (colemickens) [inline cluster/images/hyperkube/Makefile:87]: And another packaging change from the version bump? What did they do? there's a main "Cni" binary now and the plugins are distributed separately?

**Review by colemickens — CHANGES_REQUESTED** [review]

**Comment by colemickens** [comment]
(Just realized those changes are probably the same changes we've been carrying on 1.7.x for forever. If that's the case, so be it, I don't need the details, just wanted to make sure.)

File: cluster/images/hyperkube/Makefile (hunk 1 above)
Comment (rphillips) [reply cluster/images/hyperkube/Makefile:24]: this is a known version we have been using with 1.7

**Comment by rphillips** [comment]
Yeah, we've been carrying these changes for a long time.
